cmd/cortex/main.go

```diff
@@ -0,0 +1,75 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/go-kit/kit/log/level"
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/common/version"
+	"github.com/weaveworks/common/tracing"
+	"gopkg.in/yaml.v2"
+
+	"github.com/cortexproject/cortex/pkg/cortex"
+	"github.com/cortexproject/cortex/pkg/util"
+	"github.com/cortexproject/cortex/pkg/util/flagext"
+)
+
+func init() {
+	prometheus.MustRegister(version.NewCollector("loki"))
+}
+
+func main() {
+	var (
+		cfg             cortex.Config
+		configFile      = ""
+		eventSampleRate int
+	)
+	flag.StringVar(&configFile, "config.file", "", "Configuration file to load.")
+	flag.IntVar(&eventSampleRate, "event.sample-rate", 0, "How often to sample observability events (0 = never).")
+
+	flagext.RegisterFlags(&cfg)
+	flag.Parse()
+
+	if configFile != "" {
+		if err := LoadConfig(configFile, &cfg); err != nil {
+			fmt.Printf("error loading config from %s: %v\n", configFile, err)
+			os.Exit(1)
+		}
+	}
+
+	// Parse a second time, as command line flags should take precedent over the config file.
+	flag.Parse()
+
+	util.InitLogger(&cfg.Server)
+	util.InitEvents(eventSampleRate)
+
+	// Setting the environment variable JAEGER_AGENT_HOST enables tracing
+	trace := tracing.NewFromEnv(cfg.Target.String())
```

Comment:
Can we prefix it `cortex-`? It's just going to be a mess when there are other services called `ingester, distributor, etc.`

Reply:
We can!  It'll be a change, but don't think it should break too much.